tests/integration/test_mdg_generation.py

```diff
@@ -0,0 +1,464 @@
+        }
+        return [simplex_extra_args, cartesian_extra_args, tensor_grid_extra_args]
+
+    def lower_level_extra_args_data_2d(self) -> List[dict]:
+        simplex_extra_args: dict[str] = {
+            "mesh_size_min": 0.5,
+            "mesh_size_bound": 1.0,
+            "mesh_size_frac": 0.5,
+        }
+        cartesian_extra_args: dict[str] = {"nx": [10, 10], "physdims": [5, 5]}
+        tensor_grid_extra_args: dict[str] = {
+            "x": np.array([0.0, 0.5, 1.0, 1.5, 2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0]),
+            "y": np.array([0.0, 0.5, 1.0, 1.5, 2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0]),
+        }
+        return [simplex_extra_args, cartesian_extra_args, tensor_grid_extra_args]
+
+    def higher_level_extra_args_data_3d(self) -> List[dict]:
+        simplex_extra_args: dict[str] = {
+            "cell_size_min": 0.5,
+            "cell_size_bound": 1.0,
+            "cell_size_frac": 0.5,
+        }
+        cartesian_extra_args: dict[str] = {
+            "cell_size_x": 0.5,
+            "cell_size_y": 0.5,
+            "cell_size_z": 0.5,
+        }
+        tensor_grid_extra_args: dict[str] = {
+            "x_pts": np.array([0.0, 0.5, 1.0, 1.5, 2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0]),
+            "y_pts": np.array([0.0, 0.5, 1.0, 1.5, 2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0]),
+            "z_pts": np.array([0.0, 0.5, 1.0, 1.5, 2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0]),
+        }
+        return [simplex_extra_args, cartesian_extra_args, tensor_grid_extra_args]
+
+    def lower_level_extra_args_data_3d(self) -> List[dict]:
+        simplex_extra_args: dict[str] = {
+            "mesh_size_min": 0.5,
+            "mesh_size_bound": 1.0,
+            "mesh_size_frac": 0.5,
+        }
+        cartesian_extra_args: dict[str] = {"nx": [10, 10, 10], "physdims": [5, 5, 5]}
+        tensor_grid_extra_args: dict[str] = {
+            "x": np.array([0.0, 0.5, 1.0, 1.5, 2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0]),
+            "y": np.array([0.0, 0.5, 1.0, 1.5, 2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0]),
+            "z": np.array([0.0, 0.5, 1.0, 1.5, 2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0]),
+        }
+
+        return [simplex_extra_args, cartesian_extra_args, tensor_grid_extra_args]
+
+    def generate_network(self, domain_index, fracture_indices: List[int]):
+        """Construct fracture network.
+
+        Parameters:
+            domain_index (int): index of computational domain
+            fracture_indices (list): combination of fractures
+
+        Returns:
+            Union[pp.FractureNetwork2d, pp.FractureNetwork3d]: Collection of Fractures with
+            geometrical information
+        """
+
+        disjoint_fractures = None
+        domain: pp.Domain = self.domains()[domain_index]
+        if domain.dim == 2:
+            geometry = [self.fracture_2d_data()[id] for id in fracture_indices]
+            disjoint_fractures = list(map(pp.LineFracture, geometry))
```

Comment:
Please explain.

Reply:
Line 134 just maps a Callable on the list of data. However an explanation was included.